packages/wallet/src/redux/indirect-funding/player-a/__tests__/reducer.test.ts

```diff
@@ -0,0 +1,238 @@
+};
+
+const ledgerChannelDefaults = {
+  ...defaults,
+  turnNum: 5,
+  lastCommitment: {
+    commitment: testScenarios.ledgerCommitments.preFundCommitment0,
+    signature: '0x0',
+  },
+  penultimateCommitment: {
+    commitment: testScenarios.ledgerCommitments.preFundCommitment1,
+    signature: '0x0',
+  },
+  funded: false,
+  address: testScenarios.ledgerChannel.participants[0],
+  channelNonce: testScenarios.ledgerChannel.nonce,
+  libraryAddress: testScenarios.ledgerChannel.channelType,
+  participants: testScenarios.ledgerChannel.participants as [string, string],
+};
+
+const defaultAppChannelState = channelStates.waitForFundingAndPostFundSetup({
+  ...defaults,
+  turnNum: 5,
+  lastCommitment: {
+    commitment: testScenarios.preFundCommitment1,
+    signature: '0x0',
+  },
+  penultimateCommitment: {
+    commitment: testScenarios.preFundCommitment2,
+    signature: '0x0',
+  },
+  funded: false,
+  address: defaults.participants[0],
+});
+const defaultChannelState: channelStates.ChannelState = {
+  initializedChannels: {
+    [defaultAppChannelState.channelId]: defaultAppChannelState,
+  },
+  initializingChannels: {},
+};
+
+const defaultWalletState = walletStates.initialized({
+  ...testScenarios.initializedState,
+  consensusLibrary: testScenarios.ledgerLibraryAddress,
+  channelState: defaultChannelState,
+  fundingState: { directFunding: {}, indirectFunding: {} },
+  outboxState: { displayOutbox: [], messageOutbox: [], transactionOutbox: [] },
+  directFundingStore: {},
+});
+
+describe(startingIn(states.WAIT_FOR_APPROVAL), () => {
+  const { channelId } = defaults;
+  const walletState = { ...defaultWalletState };
+  walletState.indirectFunding = states.waitForApproval({ channelId });
+
+  describe(whenActionArrives(actions.indirectFunding.playerA.STRATEGY_APPROVED), () => {
+    const action = actions.indirectFunding.playerA.strategyApproved(channelId);
+    const updatedState = playerAReducer(walletState, action);
+
+    itTransitionToStateType(updatedState, states.WAIT_FOR_PRE_FUND_SETUP_1);
+    itSendsThisMessage(updatedState, MESSAGE_RELAY_REQUESTED);
+    const newLedgerId = (updatedState.indirectFunding as states.WaitForPreFundSetup1).ledgerId;
+    itTransitionsChannelToStateType(
+      updatedState,
+      newLedgerId,
+      channelStates.WAIT_FOR_PRE_FUND_SETUP,
+    );
+  });
+});
+
+describe(startingIn(states.WAIT_FOR_PRE_FUND_SETUP_1), () => {
+  const { channelId, ledgerId } = defaults;
+  const walletState = { ...defaultWalletState };
+  walletState.indirectFunding = states.waitForPreFundSetup1({ channelId, ledgerId });
+  // Add the ledger channel to state
+  const ledgerChannelState = channelStates.waitForPreFundSetup({
+    ...ledgerChannelDefaults,
+    channelId: ledgerId,
+  });
+  walletState.channelState.initializedChannels[ledgerId] = ledgerChannelState;
+
+  describe(whenActionArrives(actions.COMMITMENT_RECEIVED), () => {
+    const validateMock = jest.fn().mockReturnValue(true);
```

Comment:
Can you move this mock to the top of the test suite?

Reply:
Done